lab_5_scrapper/scrapper.py

```diff
@@ -140,10 +290,74 @@ def get_search_urls(self) -> list:
         Returns:
             list: seed_urls param
         """
+        return self.config.get_seed_urls()
 
 
 # 10
 # 4, 6, 8, 10
+class CrawlerRecursive(Crawler):
+    """
+    Recursive Crawler is a child of Crawler class.
+    """
+    def __init__(self, config: Config) -> None:
+        super().__init__(config)
+
+        self.start_url = self.base_url
+
+        self.already_crawled_path = constants.ASSETS_PATH.parent / 'already_crawled_urls.txt'
+        with open(self.already_crawled_path, 'r', encoding='utf-8') as f:
+            self.already_crawled = set(f.readlines())
+
+        self.article_tag = 'ARTICLE URL: '
+        self.urls = [url[len(self.article_tag)+1::]
+                     for url in self.already_crawled
+                     if self.article_tag in url]
```

Comment:
did not get: and if this tag is not there - we ignore the link?

Reply:
Then it's not the article page, it's other kind of page (seed, developer or other page). I save all the articles that we pass to Parser in this self.urls, and other urls are stored in another place